Makefile

```diff
@@ -0,0 +1,19 @@
+SHELL := /bin/bash
+CWD := $(shell cd -P -- '$(shell dirname -- "$0")' && pwd -P)
+
+docker-images:
+	docker build -t sphinx_rtd_theme:latest .
+
+docker-run:
+	rm -f .container_id
+	docker run --cidfile=.container_id -t -i -p 1919:1919 \
+                --network host \
+		--mount type=bind,source="$(CWD)",target=/project-readonly,readonly \
+		--mount type=volume,dst=/project-readonly/sphinx_rtd_theme.egg-info,volume-driver=local \
+		--mount type=bind,source="$(CWD)/src",target=/project/src,readonly  \
+		--mount type=bind,source="$(CWD)/docs",target=/project/docs \
+		sphinx_rtd_theme:latest $(command)
+
+docker-copy-assets:
+	docker cp "$(shell cat .container_id):/project/sphinx_rtd_theme" .
+	docker cp "$(shell cat .container_id):/project/package-lock.json" .
```

Comment:
And discussed elsewhere already, but just noting here, this homebrew configuration seems to overlap with what docker-compose provides. I'd put some preference on using standard tooling, as it's more approachable.

@humitos would have the most input/opinions here

Reply:
I don't know anything about `homebrew`, it's a MacOS thing right?

Reply:
The idea of `docker cp` is to avoid a write operation on a bind mount because file system permissions are usually troublesome, so it can be better to be explicit about what you expect to get from the docker runtime.

I'm already fixing this on your request @agjohnson

Reply:
This is now removed, the docker-compose command mounts the current git repo and builds inside that one.

Reply:
> I don't know anything about homebrew, it's a MacOS thing right?

I meant this in the sense that this is a homebrewed process -- it is a manual, one-off creation, as opposed to a standardized, repeatable process using docker-compose

Reply:
I'm not sure to understand what this step is for. Why are we copying these files from inside the container into the host?

Reply:
I had to backtrack again from the approach where a host directory was mounted and written to by the container.

The reason was that I had issues with a mix of files owned by the user and files owned by root. I also needed to run `npm cache clear` and I never understood why.

But the fundamental reason was that I was mounting host system stuff into the container and writing irrelevant files there.

The best approach IMO is to keep everything containerized and copy out the artifacts directly from their known location. If you can separate out a bind mount and ensure that ONLY intended artifacts are written there, then that's also a fine approach.

Reply:
Gotcha! Yeah, creating files _inside_ the container that then are required _outside the container_ usually generates permission problems. The only way to do that correctly is mapping the UID/GID between the host _and_ the container and that is where things get more complicated. We've been there and I didn't enjoyed it 😄 

I like the pattern you used here with `docker cp` for this requirement, where files generated inside the container are required to be distributed/used outside the container.

We currently have this problem when creating migrations on Read the Docs, for example, where the resulting `.py` files is owned by root and left in a directory owned by "your user" in the host. Then you have to run `sudo chown` to fix the permissions on that file.